### PR TITLE
Add Newdumper

### PR DESCRIPTION
**Pull request by lgrangaud**
visitors and tests of Dumper

### CODE REVIEW

**Comment by kasperosterbye** [comment]
I can see that you have a lot of failing tests. Most of those tests are in your own tests. 

If you are interested, I can do a discord session with you going through how to make sure your own test are all OK before you push to repository. We are all here to learn to be better software developers.

**Comment by lgrangaud** [comment]
> I can see that you have a lot of failing tests. Most of those tests are in your own tests.
> 
> If you are interested, I can do a discord session with you going through how to make sure your own test are all OK before you push to repository. We are all here to learn to be better software developers.

Yes, I made a mistake during the PR. So some tests crashed... I will fix it today. If I need help, I will contact you on discord. Thank you for your help

File: src/Microdown-RichTextComposer/MicRichTextComposer.class.st [hunk 1]
```diff
@@ -44,8 +44,21 @@ Class {
 	#superclass : #MicrodownVisitor,
 	#instVars : [
 		'canvas',
-		'codeStyler',
-		'textStyler'
+		'monospaceBackgroundColor',
+		'superscriptColor',
+		'subscriptColor',
```
Comment (kasperosterbye) [inline src/Microdown-RichTextComposer/MicRichTextComposer.class.st:49]: Reintroducing these instance variables in MicRichTextComposer is wrong, they were moved to the TextStyler in one of the earlier pull requests I made. Perhaps you forgot to merge those into your base before pushing?

File: src/Microdown/MicCodeBlock.class.st [hunk 2]
```diff
@@ -83,6 +83,18 @@ MicCodeBlock >> firstTagValue [
 	^ arguments at: self firstTag ifAbsent: [ '' ]
 ]
 
+{ #category : #testing }
+MicCodeBlock >> hasArguments [
+
+	^ arguments keys isNotEmpty
```
Comment (kasperosterbye) [inline src/Microdown/MicCodeBlock.class.st:89]: This works - I think you can simply ask `arguments isNotEmpty`

**Review by kasperosterbye — CHANGES_REQUESTED** [review]
The reintroduction of the many instance variables into MicRichTextComposer is wrong. They now live in the MicTextStyler class, and if you need them, take them from there, and if you need to add more, add them there.

**Pull request closed without merging**